portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadminuser/MakeARelationshipWithSystemObjectThroughAPI.testcase

```diff
@@ -0,0 +1,278 @@
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless";
+
+	setUp {
+		TestCase.setUpPortalInstanceNoSelenium();
+
+		task ("Given a Foundation object") {
+			var objectDefinitionId2 = ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_label = "foundation",
+				en_US_plural_label = "foundations",
+				name = "Foundation",
+				requiredStringFieldName = "name");
+		}
+
+		task ("Given a secondFoundation object") {
+			var objectDefinitionId3 = ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_label = "secondFoundation",
+				en_US_plural_label = "secondFoundations",
+				name = "SecondFoundation",
+				requiredStringFieldName = "name");
+		}
+
+		task ("Given manyToMany relationship usersFoundations created") {
+			var objectDefinitionId1 = ObjectDefinitionAPI.getObjectDefinitionIdByName(name = "User");
+
+			var relationshipName = ObjectDefinitionAPI.createRelationship(
+				deletionType = "cascade",
+				en_US_label = "UsersFoundations",
+				name = "usersFoundations",
+				objectDefinitionId1 = "${objectDefinitionId1}",
+				objectDefinitionId2 = "${objectDefinitionId2}",
+				type = "manyToMany");
+		}
+
+		task ("Given oneToMany relationship userSecondFoundations created") {
+			var relationshipName = ObjectDefinitionAPI.createRelationship(
+				deletionType = "cascade",
+				en_US_label = "UserSecondFoundations",
+				name = "userSecondFoundations",
+				objectDefinitionId1 = "${objectDefinitionId1}",
+				objectDefinitionId2 = "${objectDefinitionId3}",
+				type = "oneToMany");
+		}
+
+		task ("Given object entries created") {
+			ObjectDefinitionAPI.setUpGlobalObjectEntryId();
+		}
+
+		task ("Given two userAccount entries created") {
+			UserAccountAPI.setUpGlobalUserAccountId(
+				alternateName = "user1",
+				emailAddress = "user1@liferay.com",
+				familyName = "userfn1",
+				givenName = "usergn1");
+
+			UserAccountAPI.setUpGlobalUserAccountId(
+				alternateName = "user2",
+				emailAddress = "user2@liferay.com",
+				familyName = "userfn2",
+				givenName = "usergn2");
+		}
+	}
+
+	tearDown {
+		UserAccountAPI.deleteRelationship(name = "User");
+
+		for (var objectName : list "Foundation,SecondFoundation") {
+			ObjectAdmin.deleteObjectViaAPI(objectName = "${objectName}");
+		}
+
+		for (var userEmailAddress : list "user1@liferay.com,user2@liferay.com") {
+			JSONUser.deleteUserByEmailAddress(userEmailAddress = "${userEmailAddress}");
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = "4"
+	test CanDeleteCustomObjectEntryRelatedToSystemObjectByManyToMany {
+		property portal.acceptance = "true";
+
+		var responseWithUserAccount = UserAccountAPI.getRelationshipByUserAccountId();
+
+		var userAccountId1 = JSONUtil.getWithJSONPath("${responseWithUserAccount}", "$.items[?(@.givenName=='usergn1')].id");
```

Comment:
I wonder if we can make the userAccountId1 and userAccountId2 values static 🤔 @yanan @Hijky please take a look at `ObjectDefinitionAPI.setUpGlobalobjectDefinitionId();` that we use in the `portal-web/test/functional/com/liferay/portalweb/tests/enduser/headless/AdvancedObjectAPI/SupportManyToManyRelationships.testcase` and check please if we can apply it here as well :)